src/App.js

```diff
@@ -2,28 +2,37 @@ import React, { useState, useEffect } from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
-function useSemiPersistentState() {
-
-  const [todoList, setTodoList] = useState(() => {
-    const savedTodoList = localStorage.getItem("savedTodoList");
-    return savedTodoList ? JSON.parse(savedTodoList) : [];
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const loadData = new Promise ((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem("savedTodoList"))
+          }
+        })
+      }, reject, 2000);
+    });
+    loadData.then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
```

Comment:
Great work on this, I know it's a pretty complicated chunk of code. There are a couple of small issues though, and I have a suggestion based on the wording of the assignment that's a _slightly_ different approach. You can actually use the `Promise` constructor without assigning it to a variable and chain the `then` directly onto it instead of the variable. This is slightly more of a stylistic thing but it's how the instructions were worded. 

An actual issue with this code is the `reject,` on line 17. By adding that additional argument there it's breaking the `setTimeout` and the `Loading ...` doesn't show up. You've also missed adding an empty dependency array to the `useEffect`, causing it to run on every render, not just at component mount like we want. Below is an example of how it should look.

```
useEffect(() => {
    new Promise((resolve, reject) =>
      setTimeout(
        () =>
          resolve({
            data: {
              todoList: JSON.parse(localStorage.getItem('savedTodoList')) || [],
            },
          }),
        2000
      )
    ).then((result) => {
      setTodoList(result.data.todoList);
      setIsLoading(false);
    });
  }, []);
```

A tiny followup stylistic/readability suggestion is to add an empty line between the two `useEffect`s to break the code up a bit.

Reply:
got it, thank you